src/python/pants/backend/python/dependency_inference/default_module_mapping.py

```diff
@@ -3,87 +3,81 @@
 
 # NB: The project names must follow the naming scheme at
 #  https://www.python.org/dev/peps/pep-0503/#normalized-names.
+
+import re
+from functools import partial
+from typing import Match
+
+
+def all_hyphen_to_dot(m: Match) -> str:
+    return m.string.replace("-", ".")
+
+
+def first_group_hyphen_to_underscore(m: Match) -> str:
+    return m.groups()[0].replace("-", "_")
+
+
+# TODO: combine the following two functions by passing in the replacements?
+def first_group_hyphen_to_dot_second_hyphen_to_underscore(m: Match) -> str:
+    """take two groups, the first will have '-' replaced with '.', the second will have '-' replaced
+    with '_' e.g. opentelemetry-instrumentation-aio-pika ->
+    group1(opentelemtetry.instrumentation.)group2(aio_pika)"""
+    prefix = m.string[m.start(1) : m.end(1)].replace("-", ".")
+    suffix = m.string[m.start(2) : m.end(2)].replace("-", "_")
+    return f"{prefix}{suffix}"
+
+
+def two_groups_hyphen_dot_concat_with_suffix(m: Match, custom_suffix: str = "") -> str:
+    """take two groups, the first will have '-' replaced with '.', the second will have '-' replaced
+    with '' e.g. google-cloud-foo-bar -> group1(google.cloud.)group2(foobar)"""
+    prefix = m.string[m.start(1) : m.end(1)].replace("-", ".")
+    suffix = m.string[m.start(2) : m.end(2)].replace("-", "")
+    return f"{prefix}{suffix}{custom_suffix}"
+
+
+"""
+A mapping of Patterns and their replacements. will be used with `sub` e.g.
+```python
+pattern="^google-cloud-(.*)"
+replacement="google.cloud.\\g<1>" # just one '\' for realbies
+
+then if an import in the python code is google.cloud.foo, then the package of
+google-cloud-foo will be used.
+
+The match is either a string or a function that takes a re.Match and returns
+the replacement. see re.sub for more information
```

Comment:
**comment:** [docs](https://docs.python.org/3/library/re.html#re.sub) for re.sub